tmt/base.py

```diff
@@ -887,8 +888,13 @@ def root(self):
         """ Metadata root """
         return self.tree.root
 
-    def tests(self, keys=[], names=[], filters=[], conditions=[]):
+    def tests(self, keys=None, names=None, filters=None, conditions=None):
         """ Search available tests """
+        keys = keys or []
+        names = names or []
+        filters = filters or []
+        conditions = conditions or []
+
         # Apply possible command line options
         if Test._opt('names'):
             names.extend(Test._opt('names'))
```

Comment:
I think these `.extend` should be avoided too - I wouldn't expect call to `tests()` modifies its params

Reply:
Hm, that makes sense. So perhaps something like `names = names[:] or []`?